src/lib/storage/constraints/unique_checker.cpp

```diff
@@ -0,0 +1,63 @@
+#include <string>
+
+#include "storage/constraints/unique_checker.hpp"
+#include "storage/storage_manager.hpp"
+
+namespace opossum {
+
+
+bool constraint_valid_for(const Table& table, const TableConstraintDefinition& constraint,
+                          const CommitID& snapshot_commit_id, const TransactionID& our_tid) {
+  std::set<boost::container::small_vector<AllTypeVariant, 3>> unique_values;
```

Comment:
Adding a comment about that. `small_vector` because of preallocation, yes. Three slots because we think that three is the maximum number of columns most unique constraints will use.